UI/CMakeLists.txt

```diff
@@ -517,6 +517,13 @@ source_group(
 unset(_SOURCES)
 unset(_UI)
 
+get_property(OBS_MODULE_LIST GLOBAL PROPERTY OBS_MODULE_LIST)
+foreach(module ${OBS_MODULE_LIST})
+  set(SAFE_MODULES "\t\"${module}\",\n${SAFE_MODULES}")
+endforeach()
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/modules.hpp.in
```

Comment:
In general I want to remove almost all usage of generated headers in the project as they trigger rebuilds too often (the headers are regenerated by configuring the project, thus invalidating every binary that depends on them) - in this case almost the entire `obs` target will be rebuilt.

If there is a cleaner way to pass this information into the code without relying on generated files, I'd rather use that.

Reply:
The only other way I could really think of was to use a define, but having a long, quoted, and comma seprated string in a single define felt somewhat dirty/risky.

Reply:
Yeah I’d have CMake generate straight C code (so no defines), but we can only change that in bulk at a later time, similar to the version library.